src/datasets/summarizations/trend-weekly-pattern.summarization.spec.ts

```diff
@@ -0,0 +1,271 @@
+import { queryFactory } from './trend-weekly-pattern.summarization';
+import { TimeSeriesPoint } from '../metas/types';
+import { hasHighValidity, hasLowValidity, isTextPartsInSummary, norSummaryFilters } from './utils/tests';
+
+describe('queryFactory', () => {
+
+  it('should have validity value between 0 and 1.', () => {
+    const points: TimeSeriesPoint[] = [];
+    for (let i = 1; i <= 31; i++) {
+      const x = new Date(2020, 6, i);
+      const y = i;
+      points.push({ x, y });
+    }
+
+    const summaries = queryFactory(points)();
+    for (const summary of summaries) {
+      expect(summary.validity).toBeGreaterThanOrEqual(0.0);
+      expect(summary.validity).toBeLessThanOrEqual(1.0);
+    }
+  });
+
+  it('should create summaries describing monotonic increasing data.', () => {
+    const points: TimeSeriesPoint[] = [];
+    for (let i = 6; i <= 26; i++) {
+      const x = new Date(2020, 6, i);
+      const y = 200 + 5 * i;
+      points.push({ x, y });
+    }
+
+    const isMondayToSundaySummary = isTextPartsInSummary('Monday', 'Sunday', 'similar');
+    const isOtherSummary = norSummaryFilters(
+      isMondayToSundaySummary,
+    );
+
+    const summaries = queryFactory(points)();
+    const mondayToSundaySummary = summaries.filter(isMondayToSundaySummary);
+    const otherSummaries = summaries.filter(isOtherSummary);
+
+    expect(mondayToSundaySummary.length).toBe(1);
+    expect(mondayToSundaySummary.every(hasHighValidity)).toBeTrue();
+    expect(otherSummaries.every(hasLowValidity)).toBeTrue();
+  });
+
+  it('should create summaries describing monotonic decreasing data.', () => {
+    const points: TimeSeriesPoint[] = [];
+    for (let i = 6; i <= 26; i++) {
+      const x = new Date(2020, 6, i);
+      const y = 300 - 5 * i;
+      points.push({ x, y });
+    }
+
+    const isMondayToSundaySummary = isTextPartsInSummary('Monday', 'Sunday', 'similar');
+    const isOtherSummary = norSummaryFilters(
+      isMondayToSundaySummary,
+    );
+
+    const summaries = queryFactory(points)();
+    const mondayToSundaySummary = summaries.filter(isMondayToSundaySummary);
+    const otherSummaries = summaries.filter(isOtherSummary);
+
+    expect(mondayToSundaySummary.length).toBe(1);
+    expect(mondayToSundaySummary.every(hasHighValidity)).toBeTrue();
+    expect(otherSummaries.every(hasLowValidity)).toBeTrue();
+  });
+
+  it('should create summaries describing constant data.', () => {
+    const points: TimeSeriesPoint[] = [];
+    for (let i = 6; i <= 26; i++) {

```

Comment:
add a note that the test data starts on Saturday and why.

Reply:
No it starts on Monday. `new Date(2020, 6, 6)` is a Monday

Reply:
No it starts on Monday. `new Date(2020, 6, 6)` is a Monday

Reply:
Then you should add a note that it starts on Monday and why :P

Reply:
added in #107